### ISSUE TITLE
git credential helper docs & support

### ISSUE THREAD

**Issue by aran** [facebook/sapling#842]
Sapling appears to work just fine with the git credential helper protocol. For example I can configure the gcloud auth credential helper in `~/.gitconfig` and `sl clone` works. 

I can't find any reference to this functionality in the Sapling documentation. All the docs discussing similar features are specific to Github and `gh` even though the implementation appears to be more general and powerful. 

It would be nice to configure a credential helper within the repository-specific `{project}/.sl/config` rather than globally. I'm not sure if this is documented or supported.

**Comment by quark-zju** [comment]
Good point. There is a `.sl/git/config` (which is considered as impl detail) that can probably be edited to achieve what you want. Perhaps we should expose an API or config to allow editing Git configs.

**Comment by aran** [comment]
For bonus points, let me specify a credential helper inline with a flag to `sl clone`, use it for the clone operation, and configure the resulting checked out repository to use it going forward :)
